CMakeLists.txt

```diff
@@ -1,42 +1,123 @@
-set(SURGE_SKIP_LUA TRUE)
-add_subdirectory(surge)
+if ("${RACK_SDK_DIR}" STREQUAL "")
+  message(FATAL_ERROR "Path to Rack SDK is missing! Add -DRACK_SDK_DIR=<PATH> to the cmake call.")
+else ()
+  message(STATUS "use Rack-SDK in '${RACK_SDK_DIR}'")
+endif ()
+
+include(RackSDK.cmake)
 
 set(CMAKE_CXX_STANDARD 17)
 
+set(PLUGIN_NAME ${PROJECT_NAME})
+
+# Do not change the LIB_NAME!
+set(LIB_NAME plugin)
+
+find_package(Git REQUIRED)
+
+execute_process(COMMAND git rev-parse --short HEAD WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
+        OUTPUT_VARIABLE SURGE_RACK_PLUG_VERSION OUTPUT_STRIP_TRAILING_WHITESPACE)
+
+execute_process(COMMAND git rev-parse --short HEAD WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/surge
+        OUTPUT_VARIABLE SURGE_RACK_SURGE_VERSION OUTPUT_STRIP_TRAILING_WHITESPACE)
+
+set(SURGE_RACK_BASE_VERSION XT1-0-1)
+
+message(STATUS "SURGE_RACK_BASE_VERSION ${SURGE_RACK_BASE_VERSION}")
+message(STATUS "SURGE_RACK_PLUG_VERSION ${SURGE_RACK_PLUG_VERSION}")
+message(STATUS "SURGE_RACK_SURGE_VERSION ${SURGE_RACK_SURGE_VERSION}")
+
+set(SURGE_SKIP_JUCE_FOR_RACK TRUE CACHE BOOL "skip surge Juce build")
+set(SURGE_SKIP_LUA TRUE CACHE BOOL "skip surge Lua build")
+set(SURGE_SKIP_AIRWINDOWS TRUE CACHE BOOL "skip surge airwindows build")
+set(SURGE_SKIP_VST3 TRUE CACHE BOOL "skip surge VST3 build")
+set(SURGE_SKIP_ALSA TRUE CACHE BOOL "skip surge Alsa build")
+set(SURGE_SKIP_STANDALONE TRUE CACHE BOOL "skip surge standalone build")
+set(SURGE_COMPILE_BLOCK_SIZE 8 CACHE STRING "surge compile block size")
+
+add_subdirectory(surge)
+# turn off specific warning as errors enforced by Rack-SDK compiler options for surge libs
+# the warnings should be fixed in the dependencies or the warning could be disabled in the
+# dependency CMake definitions and removed from here
+set(NO_WARN_UNUSED_BUT_SET_VARIABLE -Wno-unused-but-set-variable)
+set(NO_WARN_MAYBE_UNINITIALIZED -Wno-maybe-uninitialized)
+set(NO_WARN_STRINGOP_TRUNCATION -Wno-stringop-truncation)
+set(NO_WARN_CLASS_MEMACCESS -Wno-class-memaccess)
+set(NO_WARN_INFINITE_RECURSION "")
+set(NO_WARN_UNUSED_PRIVATE_FIELD "")
+set(NO_WARN_UNUSED_LAMBDA_CAPTURE "")
+set(NO_WARN_MISMATCHED_TAGS "")
+set(NO_WARN_SELF_ASSIGN "")
+
+if (${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
+  if (CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
+    # Rack cross compiler toolchain Clang 12
+    set(NO_WARN_UNUSED_BUT_SET_VARIABLE "-Wno-unused-const-variable")
+  endif ()
+  set(NO_WARN_MAYBE_UNINITIALIZED "")
+  set(NO_WARN_STRINGOP_TRUNCATION -Wno-string-concatenation)
+  set(NO_WARN_CLASS_MEMACCESS -Wno-class-varargs)
+  set(NO_WARN_INFINITE_RECURSION -Wno-infinite-recursion)
+  set(NO_WARN_UNUSED_PRIVATE_FIELD -Wno-unused-private-field)
+  set(NO_WARN_UNUSED_LAMBDA_CAPTURE -Wno-unused-lambda-capture)
+  set(NO_WARN_MISMATCHED_TAGS -Wno-mismatched-tags)
+  set(NO_WARN_SELF_ASSIGN -Wno-self-assign)
+endif ()
+
+target_compile_options(juce_dsp_rack_sub PRIVATE -Wno-deprecated-copy ${NO_WARN_MAYBE_UNINITIALIZED})
```

Comment:
Let me know if this is OK or do you would rather disable warnings completely for some dependency libs (for 3rd party libs like sqlite.)